ledger/object/record.go

```diff
@@ -296,3 +310,114 @@ func (r *RecordDB) get(id insolar.ID) (record.Material, error) {
 
 	return rec, err
 }
+
+// RecordPositionDB is a DB storage implementation. It saves records position to DB.
+type RecordPositionDB struct {
+	lock sync.RWMutex
+	db   store.DB
+}
+
+func NewRecordPositionDB(db store.DB) *RecordPositionDB {
+	return &RecordPositionDB{db: db}
+}
+
+type recordPositionKey struct {
+	pn     insolar.PulseNumber
+	number uint32
+}
+
+func newRecordPositionKey(pn insolar.PulseNumber, number uint32) recordPositionKey {
+	return recordPositionKey{pn: pn, number: number}
+}
+
+func (k recordPositionKey) Scope() store.Scope {
+	return store.ScopeRecordPosition
+}
+
+func (k recordPositionKey) ID() []byte {
+	parsedNum := make([]byte, 4)
+	binary.BigEndian.PutUint32(parsedNum, k.number)
+	return bytes.Join([][]byte{k.pn.Bytes(), parsedNum}, nil)
+}
+
+type lastKnownRecordPositionKey struct {
+	pn insolar.PulseNumber
+}
+
+func (k lastKnownRecordPositionKey) Scope() store.Scope {
+	return store.ScopeLastKnownRecordPosition
+}
+
+func (k lastKnownRecordPositionKey) ID() []byte {
+	return bytes.Join([][]byte{k.pn.Bytes()}, nil)
+}
+
+func (r *RecordPositionDB) LastKnownPosition(pn insolar.PulseNumber) (uint32, error) {
+	r.lock.RLock()
+	defer r.lock.RUnlock()
+
+	return r.lastKnownPosition(pn)
+}
+
+func (r *RecordPositionDB) lastKnownPosition(pn insolar.PulseNumber) (uint32, error) {
+	buff, err := r.db.Get(lastKnownRecordPositionKey{pn: pn})
+	if err != nil {
+		return 0, err
+	}
+	return binary.BigEndian.Uint32(buff), nil
+}
+
+func (r *RecordPositionDB) AtPosition(pn insolar.PulseNumber, position uint32) (insolar.ID, error) {
+	r.lock.RLock()
+	defer r.lock.RUnlock()
+
+	lastKnownPosition, err := r.LastKnownPosition(pn)
+	if err != nil {
+		return insolar.ID{}, err
+	}
+	if position > lastKnownPosition {
+		return insolar.ID{}, store.ErrNotFound
+	}
+
+	positionKey := newRecordPositionKey(pn, position)
+	rawID, err := r.db.Get(positionKey)
+	if err != nil {
+		return insolar.ID{}, err
```

Comment:
wrap

Reply:
it's a base store. we don't need to wrap base errors ,such a store.ErrNotFound